src/solvers/flattening/boolbv_byte_extract.cpp

```diff
@@ -40,8 +41,12 @@ bvt map_bv(const bv_endianness_mapt &map, const bvt &src)
 
 bvt boolbvt::convert_byte_extract(const byte_extract_exprt &expr)
 {
-  // if we extract from an unbounded array, call the flattening code
-  if(is_unbounded_array(expr.op().type()))
+  // if we extract from an unbounded array or a data type that
+  // includes pointers, call the flattening code
+  if(
+    is_unbounded_array(expr.op().type()) ||
+    has_subtype(expr.type(), ID_pointer, ns) ||
+    has_subtype(expr.op().type(), ID_pointer, ns))
   {
```

Comment:
What's the rationale for that, actually?
Is the alternative to always use the flattening code?

Reply:
The rationale for lowering byte-operators when pointers are involved? Given that there is another commit that does lowering when encountering a pointer I can check whether that's still necessary (without the latter it would certainly lead to an exception), it's a matter of whether the SMT back-end now needs this or not.

Reply:
Very good call, this is now unnecessary. Commit (for both byte extract and byte update) and commit message amended.